test/integration/targets/setup_sshkey/tasks/main.yml

```diff
@@ -32,6 +32,23 @@
   tags:
     - prepare
 
+- name: create another random file
+  shell: mktemp /tmp/id_rsa.XXXXXX
+  register: another_sshkey
+  tags:
+    - prepare
+
+- name: generate another_sshkey
+  shell: echo 'y' | ssh-keygen -P '' -f {{another_sshkey.stdout}}
+  tags:
+    - prepare
+
+- name: record another key_material
+  command: cat {{another_sshkey.stdout}}.pub
```

Comment:
use a lookup rather than cat: `lookup('file', '{{ another_sshkey.stdout }}.pub')`

Reply:
in fact `another_key_material` should just be set to that lookup.